addons/mail/controllers/thread.py

```diff
@@ -125,24 +125,15 @@ def read_subscription_data(self, follower_id):
         record.check_access("read")
         # find current model subtypes, add them to a dictionary
         subtypes = record._mail_get_message_subtypes()
-        followed_subtypes_ids = set(follower.subtype_ids.ids)
-        subtypes_list = [
-            {
-                "name": subtype.name,
-                "res_model": subtype.res_model,
-                "sequence": subtype.sequence,
-                "default": subtype.default,
-                "internal": subtype.internal,
-                "followed": subtype.id in followed_subtypes_ids,
-                "parent_model": subtype.parent_id.res_model,
-                "id": subtype.id,
-            }
-            for subtype in subtypes
-        ]
-        return sorted(
-            subtypes_list,
-            key=lambda it: (it["parent_model"] or "", it["res_model"] or "", it["internal"], it["sequence"]),
```

Comment:
I believe the sort must keep a reference to parent_model though. (we display model specific subtypes before generic one - or after I don't remember which order is taken into account)